invisible_cities/reco/xy_algorithm_test.py

```diff
@@ -172,3 +172,52 @@ def test_get_nearby_sipm_inds():
             assert np.sqrt((xs[i] - xc)**2 + (ys[i] - yc)**2) <= d
         else:
             assert np.sqrt((xs[i] - xc)**2 + (ys[i] - yc)**2) >  d
+
+def test_masked_channels():
+    """
+    Scheme of SiPM positions (the numbers are the SiPM charges)
+    1 1 1
+    1 6 1
+    1 1 1
+    1 5 0
+    1 1 1
+    This test is meant to fail if either
+    1) in the case of an empty masked channel list,
+       the actual threshold in the number of SiPMs
+       around the hottest one turns out to be different from msipm
+    2) the masked channel is not taken properly into account
+       by the code
+    """
+    xs = np.array([0, 0, 0, 1, 1, 1, 2, 2, 0, 0, 1, 1, 2, 2])
+    ys = np.array([0, 1, 2, 0, 1, 2, 0, 2, 3, 4, 3, 4, 3, 4])
+    qs = np.array([1, 1, 1, 1, 5, 1, 1, 1, 1, 1, 6, 1, 1, 1])
+    pos = np.stack((xs, ys), axis=1)
+
+    nclusters = 2
+
+    xs_m = np.array([2])
+    ys_m = np.array([1])
+
+    masked_pos = np.stack((xs_m, ys_m), axis=1)
+
+    try:
```

Comment:
These `try`/`except` clauses introduce a lot of noise. The output of corona is determined unambiguously by its input and it will either run or raise an exception, but the result will be the same every time because the input doesn't change. Since you are comparing the result I assume it will always run so you can jsut simply leave the code within the `try` sections.

I would also rename `nclusters` to `expected_nclusters` (or something similar) and assign the result of `corona` to a verbose name so the comparison `assert len(something) == expected_nclusters` is more obvious.

Reply:
I've used try/except following the other examples of tests, but I can remove them.